XamarinCommunityToolkit/Views/CameraView/CameraViewRenderer.uwp.cs

```diff
@@ -0,0 +1,399 @@
+				var recommendation = videoStabilizationEffect.GetRecommendedStreamConfiguration(mediaCapture.VideoDeviceController, encodingProfile.Video);
+				if (recommendation.InputProperties != null)
+				{
+					inputPropertiesBackup = mediaCapture.VideoDeviceController.GetMediaStreamProperties(MediaStreamType.VideoRecord) as VideoEncodingProperties;
+					await mediaCapture.VideoDeviceController.SetMediaStreamPropertiesAsync(MediaStreamType.VideoRecord, recommendation.InputProperties);
+					await mediaCapture.VideoDeviceController.SetMediaStreamPropertiesAsync(MediaStreamType.VideoPreview, recommendation.InputProperties);
+				}
+				if (recommendation.OutputProperties != null)
+				{
+					outputPropertiesBackup = encodingProfile.Video;
+					encodingProfile.Video = recommendation.OutputProperties;
+				}
+				videoStabilizationEffect.Enabled = true;
+			}
+			mediaRecording = await mediaCapture.PrepareLowLagRecordToStorageFileAsync(encodingProfile, file);
+			await mediaRecording.StartAsync();
+		}
+
+		async Task<MediaSource> StopRecord()
+		{
+			if (mediaRecording == null)
+				return null;
+
+			await mediaRecording.StopAsync();
+			await mediaRecording.FinishAsync();
+			mediaRecording = null;
+
+			if (videoStabilizationEffect != null)
+			{
+				await mediaCapture.RemoveEffectAsync(videoStabilizationEffect);
+				videoStabilizationEffect = null;
+
+				if (inputPropertiesBackup != null)
+				{
+					await mediaCapture.VideoDeviceController.SetMediaStreamPropertiesAsync(MediaStreamType.VideoRecord, inputPropertiesBackup);
+					inputPropertiesBackup = null;
+				}
+				if (outputPropertiesBackup != null)
+				{
+					encodingProfile.Video = outputPropertiesBackup;
+					outputPropertiesBackup = null;
+				}
+				videoStabilizationEffect = null;
+			}
+
+			return MediaSource.FromFile(filePath);
+		}
+
+		void MediaCaptureFailed(MediaCapture sender, MediaCaptureFailedEventArgs errorEventArgs)
+			=> Element?.RaiseMediaCaptureFailed(errorEventArgs.Message);
+
+		protected override async void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			switch (e.PropertyName)
+			{
+				case nameof(CameraView.CameraOptions):
+					await CleanupCameraAsync();
+					await InitializeCameraAsync();
+					break;
+				case nameof(CameraView.FlashMode):
+					if (flash != null)
+						flash.IsEnabled = Element.FlashMode == CameraFlashMode.Torch || Element.FlashMode == CameraFlashMode.On;
+					break;
+				case nameof(CameraView.PreviewAspect):
+					// TODO
+					break;
+				case nameof(CameraView.Zoom):
+					UpdateZoom();
+					break;
+			}
+			base.OnElementPropertyChanged(sender, e);
+		}
+
+		void UpdateZoom()
+		{
+			var zoomControl = mediaCapture.VideoDeviceController.ZoomControl;
+			if (!zoomControl.Supported)
+				return;
+
+			// Added here since it's an internal method to XF
+			static float Clamp(double value, float min, float max)
```

Comment:
Can you move this to the end of the method?

Reply:
I can see, but doesn't it need to be declared before it's used?  Could just make it a private method instead of a local function.  Just wanted to keep it close incase we can remove it if we get access to XF internals.

Reply:
Good point. Well, I like to put my local functions at the end of the method... But we can move it later in another PR.

Reply:
Guess you can declare them anywhere, I'll move it to the bottom.  Didn't know you could do, guess I've always assumed they had to be declared before you can use them.